src/System.IO.FileSystem/src/System/IO/DosMatcher.cs

```diff
@@ -0,0 +1,307 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Text;
+
+namespace System.IO
+{
+    internal static class DosMatcher
+    {
+        // [MS - FSA] 2.1.4.4 Algorithm for Determining if a FileName Is in an Expression
+        // https://msdn.microsoft.com/en-us/library/ff469270.aspx
+        private static readonly char[] s_wildcardChars =
+        {
+            '\"', '<', '>', '*', '?'
+        };
+
+        /// <summary>
+        /// Change '*' and '?' to '&lt;', '&gt;' and '"' to match Win32 behavior. For compatibility, Windows
+        /// changes some wildcards to provide a closer match to historical DOS 8.3 filename matching.
+        /// </summary>
+        internal static string TranslateExpression(string expression)
+        {
+            if (string.IsNullOrEmpty(expression) || expression == "*.*")
+                return "*";
+
+            StringBuilder sb = StringBuilderCache.Acquire(expression.Length);
```

Comment:
Consider using the new ValueStringBuilder instead.

Reply:
Will do!

Reply:
Note that I'll need to make Length writable to do so- I assume that is ok.

Reply:
That's fine, assuming it's implemented correctly :)